src/TokenUriResolver.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {IJBTokenUriResolver} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBTokenUriResolver.sol";
+import {IJBProjects} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBProjects.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {JBUriOperations} from "./Libraries/JBUriOperations.sol";
+import {JBOperatable, IJBOperatorStore} from "@jbx-protocol/juice-contracts-v3/contracts/abstract/JBOperatable.sol";
+
+/**
+ * @title Juicebox TokenUriResolver Registry
+ * @notice The registry serves metadata for all Juciebox Protocol v2 projects.
+ * @dev The default metadata for all projects can be updated by the contract owner.
+ * @dev Juicebox project owners can override the default metadata for their project with their own IJBTokenUriResolver contracts.
+ */
+contract TokenUriResolver is IJBTokenUriResolver, JBOperatable, Ownable {
+    IJBProjects public immutable projects;
+    event DefaultTokenUriResolverSet(
+        IJBTokenUriResolver indexed tokenUriResolver
+    );
+    event ProjectTokenUriResolverSet(
+        uint256 indexed projectId,
+        IJBTokenUriResolver indexed tokenUriResolver
+    );
+
+    /**
+     * @notice Each project's IJBTokenUriResolver metadata contract.
+     * @dev Mapping of projectId => tokenUriResolver
+     * @dev The 0 project returns the default resolver's address.
+     * @return IJBTokenUriResolver The address of the IJBTokenUriResolver for the project, or 0 if none is set.
+     */
+    mapping(uint256 => IJBTokenUriResolver) public tokenUriResolvers;
+
+    /**
+     * @notice TokenUriResolver constructor.
+     * @dev Sets the default IJBTokenUriResolver. This resolver is used for all projects that do not have a custom resolver.
+     * @param _projects The address of the Juicebox Projects contract.
+     * @param _operatorStore The address of the JBOperatorStore contract.
+     * @param _defaultTokenUriResolver The address of the default IJBTokenUriResolver.
+     */
+    constructor(
+        IJBProjects _projects,
+        IJBOperatorStore _operatorStore,
+        IJBTokenUriResolver _defaultTokenUriResolver
+    ) JBOperatable(_operatorStore) {
+        projects = _projects;
+        tokenUriResolvers[0] = IJBTokenUriResolver(_defaultTokenUriResolver);
+    }
+
+    /**
+     *  @notice Get the token uri for a project.
+     *  @dev Called by `JBProjects.tokenUri(uint256)`. If a project has a custom IJBTokenUriResolver, it is used instead of the default resolver.
+     *  @param _projectId The id of the project.
+     *  @return tokenUri The token uri for the project.
+     *  @inheritdoc	IJBTokenUriResolver
+     */
+    function getUri(uint256 _projectId)
+        external
+        view
+        override
+        returns (string memory tokenUri)
+    {
+        IJBTokenUriResolver tur = tokenUriResolvers[_projectId];
```

Comment:
nit> 
```suggestion
        IJBTokenUriResolver _resolver = tokenUriResolvers[_projectId];
```

Reply:
Thank you, resolved [09ddd08](https://github.com/jbx-protocol/juice-token-resolver/pull/1/commits/09ddd08265164982b731cccf2b02764fc8c87344)